source/common/http/http1/codec_impl.cc

```diff
@@ -35,6 +35,13 @@ void StreamEncoderImpl::encodeHeader(const char* key, uint32_t key_size, const c
   connection_.addCharToBuffer('\n');
 }
 
+void StreamEncoderImpl::encode100ContinueHeaders(const HeaderMap& headers) {
+  ASSERT(headers.Status()->value() == "100");
+  processing_100_continue_ = true;
```

Comment:
nit: I would move the logic in `encodeHeaders` to `encodeHeadersHelper` and make it take a param of whether it's 100 continue or not, then just call the helper from both places.

Reply:
I did that in one of my clean up PRs, then broke a whole bunch of tests because of subclasses which override encodeHeaders, which is a natural thing to override, and hadn't been shifted to encodeHeadersHelper.  I thought leaving things as written was less likely to cause bugs in future but I can totally shift things around if you still prefer the refactor given additional context.

Reply:
Generally I prefer to not store what is effectively stack local state as a member variable, but it's not a big deal if it makes the code worse.